pkg/openstack/loadbalancer_test.go

```diff
@@ -593,3 +594,195 @@ func Test_getIntFromServiceAnnotation(t *testing.T) {
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+		{
+			name: "test for pool protocol UDP with proxy protocol disabled",
+			args: args{
+				protocol: "UDP",
+				svcConf: &serviceConfig{
+					keepClientIP:                true,
+					tlsContainerRef:             "tls-container-ref",
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
+					healthCheckNodePort:         2,
+					enableProxyProtocol:         false,
+				},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+							LBMethod:   "ROUND_ROBIN",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityClientIP,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "ROUND_ROBIN",
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+		{
+			name: "test for session affinity none",
+			args: args{
+				protocol: "TCP",
+				svcConf: &serviceConfig{
+					keepClientIP:                true,
+					tlsContainerRef:             "tls-container-ref",
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
+					healthCheckNodePort:         2,
+				},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+							LBMethod:   "ROUND_ROBIN",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityNone,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "ROUND_ROBIN",
+				Persistence: nil,
+			},
+		},
+		{
+			name: "test for session affinity client ip",
+			args: args{
+				protocol: "TCP",
+				svcConf: &serviceConfig{
+					keepClientIP:                true,
+					tlsContainerRef:             "tls-container-ref",
+					healthMonitorDelay:          1,
+					healthMonitorTimeout:        1,
+					healthMonitorMaxRetries:     1,
+					healthMonitorMaxRetriesDown: 1,
+					healthCheckNodePort:         2,
+				},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+							LBMethod:   "ROUND_ROBIN",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityClientIP,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "ROUND_ROBIN",
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.args.lbaasV2.buildPoolCreateOpt(tt.args.protocol, tt.args.service, tt.args.svcConf)
+			if !assert.Equal(t, got, tt.want) {
```

Comment:
You don't need to check result of `assertt.Equal()`, it'll report the issue on its own.